File: kotlin-database/src/main/java/com/xiao/database/annotation/KtTestDatabase.java
package com.xiao.database.annotation;

import com.xiao.databse.BaseDatabase;
import com.xiao.databse.annotation.KtTestDatabases;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Repeatable(KtTestDatabases.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface KtTestDatabase {
    Class<? extends BaseDatabase> database();
    Class[] mappers() default {};
}

File: kotlin-database/src/main/kotlin/com/xiao/databse/annotation/KtTestDatabase.kt
package com.xiao.databse.annotation

import com.xiao.databse.BaseDatabase
import kotlin.reflect.KClass

/**
 *  用以配置测试类使用到的数据源。
 *
 * @author lix wang
 */
@Repeatable
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class KtTestDatabase(
    val database: KClass<out BaseDatabase>,
    val mappers: Array<KClass<*>> = []
)